x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/RestrictedTrustConfig.java

```diff
@@ -7,56 +7,54 @@
 package org.elasticsearch.xpack.core.ssl;
 
 import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.core.Nullable;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.ssl.SslTrustConfig;
+import org.elasticsearch.common.ssl.StoredCertificate;
 import org.elasticsearch.common.util.CollectionUtils;
-import org.elasticsearch.env.Environment;
-import org.elasticsearch.xpack.core.ssl.cert.CertificateInfo;
-
-import javax.net.ssl.X509ExtendedTrustManager;
 
 import java.io.IOException;
 import java.nio.file.Path;
-import java.security.GeneralSecurityException;
 import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
 
+import javax.net.ssl.X509ExtendedTrustManager;
+
 /**
- * An implementation of {@link TrustConfig} that constructs a {@link RestrictedTrustManager}.
+ * An implementation of {@link SslTrustConfig} that constructs a {@link RestrictedTrustManager}.
  * This implementation always wraps another <code>TrustConfig</code> to perform the
  * underlying certificate validation.
  */
-public final class RestrictedTrustConfig extends TrustConfig {
+public final class RestrictedTrustConfig implements SslTrustConfig {
```

Comment:
I didn't put `RestrictedTrustConfig` into libs/ssl-config because it has extra dependencies, and we don't need/want to support it for SSPL code like reindex.
 
This just converts the X-Pack `RestrictedTrustConfig` to implement the ssl-config interface (`SslTrustConfig`) instead of extending the X-Pack `TrustConfig` base class (which is now deleted).